apisix/plugins/opa.lua

```diff
@@ -89,13 +90,31 @@ function _M.access(conf, ctx)
     -- parse the results of the decision
     local data, err = core.json.decode(res.body)
 
-    if err then
+    if err or not data or not data.result then
         core.log.error("invalid response body: ", res.body, " err: ", err)
```

Comment:
We should distinguish "not data or not data.result" in the error log.

Reply:
I split it into an "invalid response body", and "invalid OPA decision format: xxx err: `result` field does not exist".

Reply:
splited